storage/remote/client.go

```diff
@@ -145,41 +148,16 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 	httpReq.Header.Set("User-Agent", userAgent)
 	httpReq.Header.Set("X-Prometheus-Remote-Read-Version", "0.1.0")
 
-	ctx, cancel := context.WithTimeout(ctx, c.timeout)
-	defer cancel()
-
 	httpResp, err := c.client.Do(httpReq.WithContext(ctx))
 	if err != nil {
 		return nil, errors.Wrap(err, "error sending request")
 	}
-	defer func() {
-		io.Copy(ioutil.Discard, httpResp.Body)
-		httpResp.Body.Close()
-	}()
```

Comment:
Actually close the http body here is not an appropriate way, because STREAM should keep it open.

So for SAMPLE, I close the http body when `handleSampledResponse` return.

But for STREAM, close the http body after reading all the series or encounter error.

WDYT? ;)